corehq/apps/locations/models.py

```diff
@@ -611,6 +619,23 @@ def sql_location(self):
         notify_of_deprecation("'sql_location' was just called on a sql_location.  That's kinda silly.")
         return self
 
+    @classmethod
+    def location_and_descendants(cls, location_ids):
+        locations = []
+        for location_id in location_ids:
+            location = cls.by_location_id(location_id)
+            locations += location.get_descendants(include_self=True)
+        return locations
+
+    @classmethod
+    def location_and_descendants_ids(cls, location_ids):
+        all_locations = cls.location_and_descendants(location_ids)
+        return [location.get_id for location in all_locations]
+
+    @classmethod
+    def users_at_locations_and_descendants(cls, location_ids):
+        location_ids = SQLLocation.location_and_descendants_ids(location_ids)
+        return UserES().location(location_ids).run().hits
```

Comment:
Thanks for all the experience learning Ethan. Really appreciate it.
Managers seem to be the correct place to go for data queries. Though that overlaps a bit with dbaccessors for me.
Yet to use more of utils but i see them more as methods given data to operate on them instead of fetching data.

Moving get_locations_and_children to LocationManager from util and using that instead of adding a new method.